examples/SampleApp/src/hooks/usePaginatedPinnedMessages.ts

```diff
@@ -0,0 +1,108 @@
+import { useContext, useEffect, useRef, useState } from 'react';
+
+import { AppContext } from '../context/AppContext';
+
+import type { Channel, MessageResponse } from 'stream-chat';
+
+import type {
+  LocalAttachmentType,
+  LocalChannelType,
+  LocalCommandType,
+  LocalEventType,
+  LocalMessageType,
+  LocalReactionType,
+  LocalUserType,
+} from '../types';
+import { DEFAULT_PAGINATION_LIMIT } from '../utils/constants';
+
+export const usePaginatedPinnedMessages = (
+  channel: Channel<
+    LocalAttachmentType,
+    LocalChannelType,
+    LocalCommandType,
+    LocalEventType,
+    LocalMessageType,
+    LocalReactionType,
+    LocalUserType
+  >,
+) => {
+  const { chatClient } = useContext(AppContext);
+  const offset = useRef(0);
+  const hasMoreResults = useRef(true);
+  const queryInProgress = useRef(false);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(null);
+  const [messages, setMessages] = useState<
+    MessageResponse<
+      LocalAttachmentType,
+      LocalChannelType,
+      LocalCommandType,
+      LocalMessageType,
+      LocalReactionType,
+      LocalUserType
+    >[]
+  >([]);
+
+  const fetchPinnedMessages = async () => {
+    if (queryInProgress.current) {
+      return;
+    }
+
+    setLoading(true);
+
+    try {
+      queryInProgress.current = true;
+
+      offset.current = offset.current + messages.length;
+
+      if (!hasMoreResults.current) {
+        queryInProgress.current = false;
+        return;
+      }
+
+      // TODO: Use this when support for attachment_type is ready.
```

Comment:
should be removed, this comment is not valid here. Seems like a copy-paste leftover.

Reply:
```suggestion
```